docs/docs/step_by_step.mdx

```diff
@@ -0,0 +1,146 @@
+---
+title: Step By Step Guide
+---
+
+## Running the demo on your pc
+
+### Set environment variables
+
+```shell
+export INFRAHUB_ADDRESS="http://localhost:8000"
+export INFRAHUB_API_TOKEN="06438eb2-8019-4776-878c-0941b1f1d1ec"
+export CEOS_DOCKER_IMAGE="9r2s1098.c1.gra9.container-registry.ovh.net/external/ceos-image:4.29.0.2F"
+export LINUX_HOST_DOCKER_IMAGE="9r2s1098.c1.gra9.container-registry.ovh.net/external/alpine-host:v3.1.1"
+```
+
+### Install the Infrahub SDK
+
+```shell
+poetry install --no-interaction --no-ansi --no-root
+```
+
+### Start Infrahub
+
+```shell
+poetry run invoke start
+```
+
+### Load schema and data into Infrahub
+
+This will create :
+
+- Basics data (Account, organization, ASN, Device Type, and Tags)
+- Locations data (Locations, VLANs, and Prefixes)
+- Topology data (Topology, Topology Elements)
+- Security data (Policies, rules, objects)
+
+```shell
+poetry run invoke load-schema load-data
+```
+
+## Running the demo in Github Codespaces
+
+[Spin up in Github codespace](https://codespaces.new/opsmill/infrahub-demo-dc-fabric-develop)
+
+## Demo flow
+
+### 1. Add the repository into Infrahub (Replace GITHUB_USER and GITHUB_TOKEN)
+
+> [!NOTE]
+> Reference the [Infrahub documentation](https://docs.infrahub.app/guides/repository) for the multiple ways this can be done.
+
+```graphql
+mutation AddRepository{
+  CoreReadOnlyRepositoryCreate(
+    data: {
+      name: { value: "infrahub-demo-dc-fabric" }
+      location: { value: "https://github.com/opsmill/infrahub-demo-dc-fabric.git" }
+    }
+  ) {
+    ok
+    object {
+      id
+    }
+  }
+}
+```
+
+### 2. Generate a Topology (Device, Interfaces, Cabling, BGP Sessions, ...)
+
+> [!NOTE]
+> The example below creates the topology fra05-pod1
+
+```shell
+poetry run infrahubctl run generators/generate_topology.py topology=fra05-pod1
+```
```

Comment:
A continuation of these could be that we read this info from files instead and have a way to test that it's valid so based on the above we'd have:

### File: 0001_graphql_add_repo.gql

```graphql
mutation AddRepository{
  CoreReadOnlyRepositoryCreate(
    data: {
      name: { value: "infrahub-demo-dc-fabric" }
      location: { value: "https://github.com/opsmill/infrahub-demo-dc-fabric.git" }
    }
  ) {
    ok
    object {
      id
    }
  }
}
```

### File: 0002_shell_run_generator.sh

```shell
poetry run infrahubctl run generators/generate_topology.py topology=fra05-pod1
```

### Description

When executing these tests we'd just follow the initial number to indicate order then after that it would be file type or command type followed by some optional information that isn't required.

So if we look at `0001_graphql_add_repo.gql` only `0001_graphql_` would be required anything after that is just to help us or guide the language selection in an IDE. The second part `graphql` or `shell` would tell the test how to execute the command.

The tests themselves would need to be simple in terms of us just looking at the exit code.

Reply:
Will take on a test for `0002_shell_run_generator.sh` when this PR is done: https://github.com/opsmill/infrahub/pull/5287